### ISSUE TITLE
Enhancement: Forbid named and default imports in deferred import

### ISSUE THREAD

**Issue by fisker** [typescript-eslint/typescript-eslint#12549]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [x] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.
- [x] If I used AI assistance to prepare this issue, I have reviewed and followed the [AI Contribution Policy](https://typescript-eslint.io/contributing/ai-policy/).

### Relevant Package

typescript-estree

### My proposal is suitable for this project

- [x] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Description

The following should be invalid.

```ts
import defer x from "x";

// Default imports are not allowed in a deferred import.
```

```ts
import defer { x } from "x";

// Named imports are not allowed in a deferred import.
```
